core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.controller.ts

```diff
@@ -99,21 +103,8 @@ angular.module('oppia').directive('topicsAndSkillsDashboardPage', [
           ctrl._initDashboard = function(stayInSameTab) {
             TopicsAndSkillsDashboardBackendApiService.fetchDashboardData().then(
               function(response) {
-                // TODO(#9364): Remove the following line once this file and
-                // the corresponding spec file is upgraded to Angular 8.
-                // The following condition is required for Karma testing. The
-                // Angular HttpClient returns an Observable which when converted
-                // to a promise does not have the 'data' key but the AngularJS
-                // mocks of services using HttpClient use $http which return
-                // promise and the content is contained in the 'data' key.
-                // Therefore the following condition checks for presence of
-                // 'response.data' which would be the case in AngularJS testing
-                // but assigns 'response' if the former is not present which is
-                // the case with HttpClient.
-                var responseData = response.data ? response.data : response;
-
-                ctrl.totalTopicSummaries = responseData.topic_summary_dicts;
-                ctrl.topicSummaries = responseData.topic_summary_dicts;
+                ctrl.totalTopicSummaries = response.topic_summary_dicts;
+                ctrl.topicSummaries = response.topic_summary_dicts;
```

Comment:
Why are you setting  these two vars to the same thing? Just use one.

Reply:
I need three arrays for the pagination, first one is the complete original one, second is the filtered one(based on the filters applied) and third one is the array that is being displayed which is a subset of the second array based on page number and items per page.